### PR TITLE
Add support for GPytorch

### PR DESCRIPTION
**Pull request by felipeZ**
## changes
* Solve #90  by adding interface to [GPytorch](https://gpytorch.ai/)

### CODE REVIEW

File: scripts/run_torch_models.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from pathlib import Path
 import torch
 from swan.dataset import TorchGeometricGraphData, FingerprintsData, DGLGraphData
-from swan.modeller import Modeller
+from swan.modeller import TorchModeller
```
Comment (NicoRenaud) [inline scripts/run_torch_models.py:7]: Is the TorchModeller valid for any pytorch model or only GPyTorch models ?

**Review by NicoRenaud — APPROVED** [review]
looks good to me but it's difficult to say more without further knowledge of gpytorch

File: scripts/run_torch_models.py (hunk 1 above)
Comment (felipeZ) [reply scripts/run_torch_models.py:7]: `TorchModeller` represents all the PyTorch models expect for the `GPyTorch` models

File: swan/dataset/splitter.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+T_co = TypeVar('T_co', bound=Union[np.ndarray, torch.Tensor], covariant=True)
+
+
+class SplitDataset(NamedTuple, Generic[T_co]):
+    indices: np.ndarray       # Shuffled indices to split the data
+    ntrain: int               # Number of points used for training
+    features_trainset: T_co   # Features for training
+    features_validset: T_co   # Features for validation
+    labels_trainset: T_co     # Labels for training
+    labels_validset: T_co     # Labels for validation
```
[the remark this answers is not in the document]
Comment (felipeZ) [reply swan/dataset/splitter.py:15]: deep sigh...

**Pull request merged**